### PR TITLE
Rebase to 3.17, migrate to s6v3

### PR DESCRIPTION
**Pull request by thespad**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-pylon/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:
Changed order of operations so we're not recursively chowning the /code folder on every start - only do it now if we've cloned a git repo (as root), otherwise do a non-recursive chown. Prevents us screwing up host permissions on people's code if they mount in their own stuff.

### CODE REVIEW

File: root/etc/s6-overlay/s6-rc.d/svc-pylon/run [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+          node server.js -l 0.0.0.0 -p 3131 -w /code \
+          --username "${PYUSER}" --password "${PYPASS}"
+fi
+
+if [[ -z ${PYUSER+x} ]] && [[ -z ${PYPASS+x} ]]; then
+    exec \
+        s6-notifyoncheck -d -n 300 -w 1000 -c "nc -z localhost 3131" \
+          cd /app/pylon s6-setuidgid abc \
+          node server.js -l 0.0.0.0 -p 3131 -w /code
+fi
```
Comment (aptalca) [inline root/etc/s6-overlay/s6-rc.d/svc-pylon/run:22]: What if one's set and not the other? :-P

Comment (thespad) [reply root/etc/s6-overlay/s6-rc.d/svc-pylon/run:22]: Good question. I just copied the existing logic, but we should probably have a catch for it.

Comment (thespad) [reply root/etc/s6-overlay/s6-rc.d/svc-pylon/run:22]: Fixed in bf3ab6f, starts without user/pass (and logs) if only one or the other is set.

Comment (aptalca) [reply root/etc/s6-overlay/s6-rc.d/svc-pylon/run:22]: looks great

**Review by aptalca — APPROVED** [review]

**Pull request merged**